src/Documenter.jl

```diff
@@ -624,6 +621,29 @@ function rm_and_add_symlink(target, link)
     symlink(target, link)
 end
 
+"""
+    user_host_upstream(repo)
+
+Disassemble repo address into user, host, and path to repo. If no user is given, default to
+"git". Reassemble user, host and path into an upstream to `git push` to. 
+"""
+function user_host_upstream(repo)
+    #= the regex has three parts:
+    (?:([^@]*)@)?  matches any number of characters up to the first "@", if present,
+        capturing only the characters before the "@" - this captures the username
+    (?:([^\/:]*)[\/:]){1}  matches exactly one instance of any number of characters
+        other than "/" or ":" up to the first "/" or ":" - this captures the hostname
+    [\/]?(.*)  matches the rest of the repo, except an initial "/" if present (e.g. if
+        repo is of the form usr@host:/path/to/repo) - this captures the path on the host
+    =#
+    m = match(r"(?:([^@]*)@)?(?:([^\/:]*)[\/:]){1}[\/]?(.*)", repo)
+    isnothing(m) && error("Invalid repo path $repo")
```

Comment:
This is failing on 1.0 (we're not depending on Compat):

```suggestion
    (m === nothing) && error("Invalid repo path $repo")
```